mobileapps/views.py

```diff
@@ -296,3 +302,187 @@ def delete(self, request, mobileapp_id):
+            publish_mobile_apps_notifications_task.delay(user_ids, notification_message, api_keys,
+                                                         notification_provider)
+
+        except Exception, ex:
+            return Response({'message':  _('Server error')}, status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+        return Response({'message': _('Accepted')}, status.HTTP_202_ACCEPTED)
+
+
+class MobileAppOrganizationAllUsersNotifications(SecureAPIView):
+    """
+    **Use Cases**
+
+        send a push notification to all the users registered with the specific
+        organization of the app using app's push notifications provider.
+
+    **Example Requests**
+
+        POST /api/mobileapps/{id}/organization/{id}/notification
+
+        The body of the POST request must include the following parameters.
+
+        * message: notification message
+
+    **Response Values**
+
+        If the request is successful, the request returns an HTTP 202 "Accepted" response.
+
+        The HTTP 202 response has the following value.
+
+        * message: Accepted
+    """
+
+    def post(self, request, pk, organization_id):
+
+        message = request.data.get('message', None)
+        if not message:
+            return Response({'message': _('message is missing')}, status.HTTP_400_BAD_REQUEST)
+
+        try:
+            mobile_app = MobileApp.objects.get(pk=pk)
+            if not mobile_app.is_active:
+                return Response({'message': _('Mobile app is inactive')}, status.HTTP_403_FORBIDDEN)
+        except ObjectDoesNotExist:
+            return Response({'message': _('Mobile app does not exist')}, status.HTTP_404_NOT_FOUND)
+
+        try:
+            organization = mobile_app.organizations.get(id=organization_id)
+        except ObjectDoesNotExist:
+            return Response({'message': _('Organization is not associated with mobile app')},
+                            status.HTTP_400_BAD_REQUEST)
+        try:
+            user_ids = organization.users.values_list('id', flat=True).all()
+            api_keys = mobile_app.get_api_keys()
+            notification_provider = mobile_app.get_notifications_provider()
+            if not notification_provider:
+                return Response({'message': _('Notification Provider not found')}, status.HTTP_404_NOT_FOUND)
+            notification_message = _create_notification_message(message, mobile_app.identifier)
+
+            # Send the notification_msg to the Celery task
+            publish_mobile_apps_notifications_task.delay(user_ids, notification_message, api_keys,
+                                                         notification_provider)
+
+        except Exception, ex:  # pylint: disable=broad-except
+            return Response({'message':  _('Server error')}, status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+        return Response({'message': _('Accepted')}, status.HTTP_202_ACCEPTED)
+
+
+def _create_notification_message(title, app_identifier, send_to_all=False):
```

Comment:
This method should have two params `app_identifier ` and `payload`. It would make it more generic we can pass whatever payload we want to.